src/lib/core/components/visualizations/InputVariables.tsx

```diff
@@ -17,6 +17,7 @@ import {
 interface InputSpec {
   name: string;
   label: string;
+  wrapper?: React.FunctionComponent;
```

Comment:
I think this approach of adding a "wrapper" to the `InputSpec` could work!

It might be more idiomatic to express the wrapping as `InputComponent: React.ComponentType<InputProps>` or `renderInput: (props: InputProps) => React.ReactNode` on the `InputSpec`.

(Here, `InputProps` is the current props interface for the `VariableTreeDropdown`.)

One benefit of this approach is a wrapper will have access to the `InputProps` passed via `InputVariables`, which the wrapper might need as part of its custom rendering.